frontend/src/TableLayout.tsx

```diff
@@ -72,35 +73,34 @@ export default function TableLayout({
 
             {error && <ErrorDisplay message={error} sx={{ pb: 2 }} />}
 
-            {loading ? (
-                "Loading..."
-            ) : (
-                <Box
+            <Box
+                sx={{
+                    overflow: loading ? "hidden" : "auto",
+                    boxShadow: "lg",
+                    borderRadius: "sm",
+                    position: "relative",
+                    ...containerStyle,
+                }}
+            >
+                {loading && <LoadingOverlay />}
```

Comment:
Line 81 and line 85 are new - the rest in this file is just indent change from moving line 75 -77 `loading` handling inside the `Box`